osu.Game/Overlays/SkinEditor/SkinEditorChangeHandler.cs

```diff
@@ -56,20 +57,56 @@ protected override void ApplyStateChange(byte[] previousState, byte[] newState)
             if (deserializedContent == null)
                 return;
 
-            SerialisedDrawableInfo[] skinnableInfo = deserializedContent.ToArray();
-            Drawable[] targetComponents = firstTarget.Components.OfType<Drawable>().ToArray();
+            SerialisedDrawableInfo[] skinnableInfos = deserializedContent.ToArray();
+            ISerialisableDrawable[] targetComponents = firstTarget.Components.ToArray();
 
-            if (!skinnableInfo.Select(s => s.Type).SequenceEqual(targetComponents.Select(d => d.GetType())))
+            // Store components based on type for later lookup
+            var typedComponents = new Dictionary<Type, Stack<Drawable>>();
+
+            for (int i = targetComponents.Length - 1; i >= 0; i--)
```

Comment:
Is this indexing backwards because of the stack inverting order? If so, then that I didn't foresee, but you could fix this by using `Queue<T>` which is FIFO rather than `Stack<T>` which is LIFO.